src/server/pegasus_server_impl.cpp

```diff
@@ -1198,7 +1198,8 @@ void pegasus_server_impl::on_get_scanner(get_scanner_rpc rpc)
             epoch_now,
             request.no_value,
             request.__isset.validate_partition_hash ? request.validate_partition_hash : true,
-            return_expire_ts);
+            return_expire_ts,
+            !request.only_return_count);
```

Comment:
I found you reverse the value every calls of append_key_value_for_scan, why not use it directly?